src/authorizations/components/EditTokenOverlay.tsx

```diff
@@ -61,7 +61,7 @@ const EditTokenOverlay: FC<Props> = props => {
     if (_.isEmpty(permissions)) {
       formatPermissions()
     }
-  }, [])
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
 
```

Comment:
Same here - if the dependency we're waiting on is `formatPermissions()`, it's likely safe to include in the array since it shouldn't change. It is definitely worth re-testing the page after adding the dependencies though, just to make sure it doesn't produce any unexpected behavior.

Reply:
here, `permissions` and `formatPermissions` is what eslint is expecting in the dependency array. To add formatPermissions to the array, I'll have to restructure the code and move the useEffect hook after `formatPermissions` function is declared. Doing that will make the dependencies of useEffect Hook change on every render. To fix this issue, we can move the function inside the useEffect callback. or wrap the definition of 'formatPermissions' in its own useCallback() Hook. I tried the second option wrapping it in its own useCallback hook and it worked fine, but... is all of this necessary? @hoorayimhelping do you have opinions on this?

Reply:
Since `formatPermissions` uses props, I think we'll want it to be part of the dependency array, since when the props change, the outcome of the function is liable to change as well. Seems like doing the `useCallback` treatment is the best idea here

Reply:
okay, after reading [this](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies), I see why it's necessary @wdoconnell. and thank you Bucky for chiming in.